mod/wikirate/set/abstract/import.rb

```diff
@@ -37,14 +49,28 @@ def construct_value_args args
   create_args
 end
 
+def check_duplication_with_existing metric_value_name, source_card
+  slot_args = success.slot
+  if (source = Card[metric_value_name.to_name.field(:source)])
+    if source.item_cards[0].key == source_card.key
+      slot_args[:identical_metric_value].push(metric_value_name)
+    else
+      slot_args[:duplicated_metric_value].push(metric_value_name)
+    end
+    return true
+  end
+  false
+end
+
 # @return updated or created metric value card object
 def parse_metric_value import_data, source_map
   args = process_metric_value_data import_data
   process_source args, source_map
   ensure_company_exists args[:company]
   return unless valid_value_data? args
   return unless (create_args = construct_value_args args)
-  check_duplication create_args[:name], args[:row]
+  check_duplication_in_subcards create_args[:name], args[:row]
+  return if check_duplication_with_existing create_args[:name], args[:source]
```

Comment:
these two have the same naming pattern, but one returns and one doesn't.  pls comment and explain